AK Zeus/app/src/main/java/ayds/zeus/songinfo/moredetails/model/repository/broker/proxies/LastFMProxy.kt

```diff
@@ -0,0 +1,25 @@
+package ayds.zeus.songinfo.moredetails.model.repository.broker.proxies
+
+import ayds.apolo2.LastFM.LastFMAPIArtistService
+import ayds.apolo2.LastFM.entities.InfoArtist
+import ayds.zeus.songinfo.moredetails.model.entities.Card
+import ayds.zeus.songinfo.moredetails.model.entities.EmptyCard
+import ayds.zeus.songinfo.moredetails.model.entities.Source
+
+internal class LastFMProxy(private val service: LastFMAPIArtistService) : Proxy {
+
+    override fun getCard(artistName: String): Card {
+        val artistInfo = service.getArtist(artistName)
+        return if (artistInfo == null)
+            EmptyCard()
+        else
+            map(artistInfo)
+    }
+
+    private fun map(article: InfoArtist) = Card(
+        article.description,
+        article.infoUrl,
+        article.sourceLogoUrl,
+        Source.LASTFM
+    )
```

Comment:
Comentario menor, el mapeo podría mapear la emptyCard
```suggestion
    override fun getCard(artistName: String): Card = service.getArtist(artistName).toCard()

    private fun InfoArtist?.toCard() =
        when {
            this == null -> EmptyCard()
            else -> Card(description, infoUrl, sourceLogoUrl, Source.LASTFM)
        }
```